### PR TITLE
8282511: Use fixed certificate validation date in SSLExampleCert template

### PR DESCRIPTION
**Pull request by XueleiFan**
May I have this test update reviewed?

The certificates used in SSL testing template SSLExampleCert could expired in the future (for example [JDK-8282398](https://bugs.openjdk.java.net/browse/JDK-8282398)).  It is not always easy to replace the certificates if the template has been used a lot.

This update is trying to use a fixed validation date, so that even the certificates are expired the test will continue to work with no surprises.

I also did a few code clean up in this patch.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8282511](https://bugs.openjdk.java.net/browse/JDK-8282511): Use fixed certificate validation date in SSLExampleCert template

### Reviewers
 * [Rajan Halade](https://openjdk.java.net/census#rhalade) (@rhalade - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7651/head:pull/7651` \
`$ git checkout pull/7651`

Update a local copy of the PR: \
`$ git checkout pull/7651` \
`$ git pull https://git.openjdk.java.net/jdk pull/7651/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7651`

View PR using the GUI difftool: \
`$ git pr show -t 7651`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7651.diff">https://git.openjdk.java.net/jdk/pull/7651.diff</a>

</details>

### CODE REVIEW

File: test/jdk/javax/net/ssl/ServerName/EndingDotHostname.java [hunk 1]
```diff
@@ -23,10 +23,11 @@
 
 /**
  * @test
- * @bug 8065422
+ * @bug 8806542 8282511
```
Comment (rhalade) [inline test/jdk/javax/net/ssl/ServerName/EndingDotHostname.java:26]: No need to add a bug id here as this is test only change

File: test/jdk/javax/net/ssl/templates/SSLExampleCert.java [hunk 2]
```diff
@@ -359,18 +368,41 @@ private static SSLContext createSSLContext(
             }
         }
 
+        // Set the date for the verifying of certificates.
+        @SuppressWarnings("deprecation")
+        Date verifyingDate = new Date(123, 3, 3);
```
Comment (rhalade) [inline test/jdk/javax/net/ssl/templates/SSLExampleCert.java:373]: can DateFormat be used instead to parse Date from a readable string representation?

File: test/jdk/javax/net/ssl/ServerName/EndingDotHostname.java (hunk 1 above)
Comment (XueleiFan) [reply test/jdk/javax/net/ssl/ServerName/EndingDotHostname.java]: Yep.

File: test/jdk/javax/net/ssl/templates/SSLExampleCert.java (hunk 2 above)
Comment (XueleiFan) [reply test/jdk/javax/net/ssl/templates/SSLExampleCert.java]: Good catch!  Thank you!

File: test/jdk/javax/net/ssl/ServerName/EndingDotHostname.java [hunk 3]
```diff
@@ -23,10 +23,11 @@
 
 /**
  * @test
- * @bug 8065422
+ * @bug 8806542
```
[the remark this answers is not in the document]
Comment (XueleiFan) [reply test/jdk/javax/net/ssl/ServerName/EndingDotHostname.java]: Oops, definitely not an expected update.  Thank you for the nice catch.  Updated.

**Review by rhalade — APPROVED** [review]

**Pull request closed without merging**